Ginger/GingerCoreNET/RunLib/DynamicExecutionLib/DynamicExecutionManager.cs

```diff
@@ -369,24 +371,579 @@ public static void CreateRealRunSetFromDynamic(RunsetExecutor runsetExecutor, Ad
+                                                new Tuple<string, Guid?>(nameof(RunSetActionBase.Guid), runsetOperationConfigMail.ID),
+                                                new Tuple<string, string>(nameof(RunSetActionBase.Name), runsetOperationConfigMail.Name),
+                                                runSetConfig.RunSetActions);
+                        }
+                        else
+                        {
+                            mailOperation = new RunSetActionHTMLReportSendEmail();
+                            mailOperation.Name = runsetOperationConfigMail.Name;
+                            mailOperation.HTMLReportTemplate = RunSetActionHTMLReportSendEmail.eHTMLReportTemplate.HTMLReport;
+                            mailOperation.selectedHTMLReportTemplateID = 100;//ID to mark defualt template
+                            mailOperation.Email.IsBodyHTML = true;
+                            mailOperation.Condition = RunSetActionBase.eRunSetActionCondition.AlwaysRun;
+                            mailOperation.RunAt = RunSetActionBase.eRunAt.ExecutionEnd;
+                        }                        
+
+                        if (runsetOperationConfigMail.MailSettings.EmailMethod != null)
+                        {
+                            if (runsetOperationConfigMail.MailSettings.EmailMethod == SendMailSettings.eEmailMethod.OUTLOOK)
+                            {
+                                mailOperation.Email.EmailMethod = GingerCore.GeneralLib.Email.eEmailMethod.OUTLOOK;
+                            }
+                            else
+                            {
+                                mailOperation.Email.EmailMethod = GingerCore.GeneralLib.Email.eEmailMethod.SMTP;
+                                if (runsetOperationConfigMail.MailSettings.SmtpDetails != null)
+                                {
+                                    if (runsetOperationConfigMail.MailSettings.SmtpDetails.Server != null)
+                                    {
+                                        mailOperation.Email.SMTPMailHost = runsetOperationConfigMail.MailSettings.SmtpDetails.Server;
+                                    }
+                                    if (runsetOperationConfigMail.MailSettings.SmtpDetails.Port != null)
+                                    {
+                                        mailOperation.Email.SMTPPort = int.Parse(runsetOperationConfigMail.MailSettings.SmtpDetails.Port);
+                                    }
+                                    if (runsetOperationConfigMail.MailSettings.SmtpDetails.EnableSSL != null)
+                                    {
+                                        mailOperation.Email.EnableSSL = (bool)runsetOperationConfigMail.MailSettings.SmtpDetails.EnableSSL;
+                                    }
+                                    if (string.IsNullOrEmpty(runsetOperationConfigMail.MailSettings.SmtpDetails.User) == false)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=4812633)